src/qutip_qip/circuit/text_renderer.py

```diff
@@ -0,0 +1,497 @@
+"""
+Module for rendering a quantum circuit in text format.
+"""
+
+from math import ceil
+from typing import List
+
+from .base_renderer import BaseRenderer, StyleConfig
+from ..operations import Gate, Measurement
+from . import QubitCircuit
+
+__all__ = [
+    "TextRenderer",
+]
+
+
+class TextRenderer(BaseRenderer):
+    """
+    A class to render a quantum circuit in text format.
+    """
+
+    def __init__(self, qc: QubitCircuit, **style):
+
+        # user defined style
+        style = {} if style is None else style
+        style["gate_margin"] = 0
+        self.style = StyleConfig(**style)
+
+        super().__init__(self.style)
+        self._qc = qc
+        self._qwires = qc.N
+        self._cwires = qc.num_cbits
+        self._layer_list = {i: [] for i in range(self._qwires + self._cwires)}
+
+        self._render_strs = {
+            "top_frame": ["  "] * (self._qwires + self._cwires),
+            "mid_frame": ["──"] * self._qwires + ["══"] * self._cwires,
+            "bot_frame": ["  "] * (self._qwires + self._cwires),
+        }
+
+    def _adjust_layer(self, wire_list: List[int], xskip: int):
+        """
+        Adjust the layers by filling the empty spaces with respective characters
+        """
+
+        for wire in wire_list:
+            self._render_strs["top_frame"][wire] += " " * (
+                xskip - len(self._render_strs["top_frame"][wire])
+            )
+            self._render_strs["bot_frame"][wire] += " " * (
+                xskip - len(self._render_strs["bot_frame"][wire])
+            )
+            if wire < self._qwires:
+                self._render_strs["mid_frame"][wire] += "─" * (
+                    xskip - len(self._render_strs["mid_frame"][wire])
+                )
+            else:
+                self._render_strs["mid_frame"][wire] += "═" * (
+                    xskip - len(self._render_strs["mid_frame"][wire])
+                )
```

Comment:
Since we're dealing with strings and the gates vary in width, `_adjust_layer` adds padding to maintain consistent layer alignment across the different wires. The function `_add_wire_all` appears to be of different menaing, but to clarify its purpose, renaming it to `_adjust_layer_pad` would make its role clearer.